myst_parser/main.py

```diff
@@ -1,72 +1,118 @@
+    dmath_allow_digits: bool = attr.ib(default=True, validator=instance_of(bool))
+    amsmath_enable: bool = attr.ib(default=False, validator=instance_of(bool))
+
+    override_mathjax: bool = attr.ib(default=True, validator=instance_of(bool))
+
+    admonition_enable: bool = attr.ib(default=False, validator=instance_of(bool))
+
+    disable_syntax: List[str] = attr.ib(
+        factory=list,
+        validator=deep_iterable(instance_of(str), instance_of((list, tuple))),
+    )
+
+    html_img_enable: bool = attr.ib(default=False, validator=instance_of(bool))
+
+    # see https://en.wikipedia.org/wiki/List_of_URI_schemes
+    url_schemes: Optional[List[str]] = attr.ib(
+        default=None,
+        validator=optional(deep_iterable(instance_of(str), instance_of((list, tuple)))),
+    )
+
+    def as_dict(self, dict_factory=dict) -> dict:
+        return attr.asdict(self, dict_factory=dict_factory)
+
+
+def default_parser(config: MdParserConfig) -> MarkdownIt:
     """Return the default parser configuration for MyST"""
-    if renderer == "sphinx":
+    if config.renderer == "sphinx":
         from myst_parser.sphinx_renderer import SphinxRenderer
 
         renderer_cls = SphinxRenderer
-    elif renderer == "html":
+    elif config.renderer == "html":
         renderer_cls = RendererHTML
-    elif renderer == "docutils":
+    elif config.renderer == "docutils":
         from myst_parser.docutils_renderer import DocutilsRenderer
 
         renderer_cls = DocutilsRenderer
     else:
-        raise ValueError("unknown renderer type: {0}".format(renderer))
+        raise ValueError("unknown renderer type: {0}".format(config.renderer))
 
     md = (
         MarkdownIt("commonmark", renderer_cls=renderer_cls)
         .enable("table")
         .use(front_matter_plugin)
         .use(myst_block_plugin)
         .use(myst_role_plugin)
-        .use(texmath_plugin, delimiters=math_delimiters)
         .use(footnote_plugin)
         .disable("footnote_inline")
         # disable this for now, because it need a new implementation in the renderer
         .disable("footnote_tail")
+    )
+    if config.dmath_enable:
+        md.use(
+            dollarmath_plugin,
+            allow_labels=config.dmath_allow_labels,
+            allow_space=config.dmath_allow_space,
+            allow_digits=config.dmath_allow_digits,
+        )
+    if config.admonition_enable:
         # we don't want to yet remove un-referenced, because they may be referenced
         # in admonition type directives
         # so we do our own post processing
-    )
-    for name in disable_syntax:
-        md.disable(name, True)
-    if enable_admonitions:
         md.use(container_plugin, "admonition", validate=validate_admonition)
-    if enable_amsmath:
+    if config.amsmath_enable:
         md.use(amsmath_plugin)
+    for name in config.disable_syntax:
+        md.disable(name, True)
+
+    md.options.update(
+        {
+            "enable_html_img": config.html_img_enable,
```

Comment:
@chrisjsewell is there a reason to have different option names here?

Reply:
Not really, I just did want to make a breaking change to the internal code. But yeh can be changed at some point